code/modules/xenomorph/xeno_structures.dm

```diff
@@ -127,36 +119,21 @@
 	COOLDOWN_START(src, silo_damage_alert_cooldown, XENO_HEALTH_ALERT_COOLDOWN) //set the cooldown.
 
 ///Alerts the Hive when hostiles get too close to their resin silo
-/obj/structure/resin/silo/proc/resin_silo_proxy_alert(datum/source, atom/hostile)
-	SIGNAL_HANDLER
-
+/obj/structure/resin/silo/proc/resin_silo_proxy_alert()
 	if(!COOLDOWN_CHECK(src, silo_proxy_alert_cooldown)) //Proxy alert triggered too recently; abort
 		return
-
-	if(!isliving(hostile))
-		return
-
-	var/mob/living/living_triggerer = hostile
-	if(living_triggerer.stat == DEAD) //We don't care about the dead
-		return
-
-	if(isxeno(hostile))
-		var/mob/living/carbon/xenomorph/X = hostile
-		if(X.hive == associated_hive) //Trigger proxy alert only for hostile xenos
-			return
```

Comment:
New code doesn't detect hostile hives.

Reply:
Yes, but HVH sucks. :( lazy me didn't want to add

Reply:
I think you can do similar loop using `cheap_get_xenos_near`, adding different hive check in `if` condition.

Reply:
yes i know, but i'm lazy ahah